resolver/resolver.cc

```diff
@@ -3683,6 +3683,79 @@ class ResolveSanityCheckWalk {
     }
 };
 
+class CheckAbstractClassInitialization {
+public:
+    struct CheckAbstractClassInitializationWalkResult {
+        vector<ast::ParsedFile> trees;
+    };
+
+    ast::ExpressionPtr postTransformSend(core::Context ctx, ast::ExpressionPtr tree) {
+        auto &send = ast::cast_tree_nonnull<ast::Send>(tree);
+        if (send.fun != core::Names::new_()) {
+            return tree;
+        }
+
+        auto *id = ast::cast_tree<ast::ConstantLit>(send.recv);
+        if (id == nullptr || !id->symbol.exists() || !id->symbol.isClassOrModule()) {
+            return tree;
+        }
+
+        if (id->symbol == core::Symbols::StubModule()) {
```

Comment:
Isn't this already covered by `id->symbol.exists()` checked earlier?

Reply:
I don't have a lot of context on stub modules - are stub modules not included in the list of `exists()`?

Reply:
I just checked and `core::Symbols::StubModule().exists() == true` so we may need it.

In which case do we get a StubModule here?